### PR TITLE
OWLS-103358, OWLS-106012 - fixed intermittent failures and adapted wdt3.0 changes

### PR DESCRIPTION
**Pull request by marinakog**
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/15764

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 1]
```diff
@@ -733,40 +738,92 @@ void testCreateDomainWithConfigMapAndEmptryModelFileDir() {
...
             miiAuxiliaryImage12);
+    assertNotNull(domainCR1, "failed to create domain resource");
+    domainCR1.spec().configuration().model().configMap(configMapName1);
+    createDomainAndVerify(domainUid, domainCR1, domainNamespace,
+            adminServerPodName, managedServerPrefix, replicaCount);
+  }
+
+  /**
+   * Create a domain using auxiliary image with empty configMap and no model files.
+   * Verify domain events and operator log contains the expected error message.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:770]: Provide the description of expected message  (createDomain argument processing failed: createDomain failed to find a model file)

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 2]
```diff
@@ -708,15 +714,14 @@ WEBLOGIC_IMAGE_TO_USE_IN_SPEC, adminSecretName, createSecretsForImageRepos(domai
...
+   * Create a domain using auxiliary image with configMap with model files.
    * Verify domain is created and running.
    */
   @Test
-  @DisplayName("Test to create domain using auxiliary image with configMap and empty model"
-          + " files dir in the auxiliary image")
-  void testCreateDomainWithConfigMapAndEmptryModelFileDir() {
+  @DisplayName("Test to create domain using auxiliary image with configMap containing model files"
+      + " with empty model"
+      + " files dir in the auxiliary image")
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:723]: can you combines these two lines into one?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java [hunk 3]
```diff
@@ -779,30 +836,15 @@ WEBLOGIC_IMAGE_TO_USE_IN_SPEC, adminSecretName, createSecretsForImageRepos(domai
...
-    domainCR1.spec().configuration().model().configMap(configMapName1);
-    createDomainAndVerify(domainUid, domainCR1, domainNamespace,
-            adminServerPodName, managedServerPrefix, replicaCount);
+    deleteConfigMap(configMapName, domainNamespace);
+    testUntil(
+        withLongRetryPolicy,
+        () -> listConfigMaps(domainNamespace).getItems().stream().noneMatch((cm)
+            -> (cm.getMetadata().getName().equals(configMapName))),
+        logger,
+        "configmap {0} to be deleted.");
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAuxiliaryImage.java:847]: You need to add configmap here.
testUntil(
         withLongRetryPolicy,
         () -> listConfigMaps(domainNamespace).getItems().stream().noneMatch((cm)
             -> (cm.getMetadata().getName().equals(configMapName))),
         logger,
         "configmap {0} to be deleted",
         configMapName);

**Review by xiancao — CHANGES_REQUESTED** [review]

**Review by anpanigr — COMMENTED** [review]
Make sure all changes are backported into release/4.0 and release/3/.4

**Review by xiancao — APPROVED** [review]

**Pull request merged**